File: lokerpc/codegen/testdata/spaces-optional.json.ts
import { RPCContextClient } from "@loke/http-rpc-client";
import { Context } from "@loke/context";

export type AccountMetadata = {
  Environment: string;
  "Operator URL": string | null;
  "Organization ID": string;
  "Organization Name": string;
  "Location ID"?: string | null;
  "Location Name"?: string | null;
};

/**
 * Test service for AccountMetadata shape
 */
export class StripePaymentsService extends RPCContextClient {
  constructor(baseUrl: string) {
    super(baseUrl, "stripe-payments")
  }
  /**
   * Fetch account metadata
   */
  getAccountMetadata(ctx: Context, req: AccountMetadata): Promise<AccountMetadata> {
    return this.request(ctx, "getAccountMetadata", req);
  }
}

File: lokerpc/codegen/testdata/spaces-optional.json.go
package stripepayments

import (
	"context"

	"github.com/LOKE/pkg/lokerpc"
)

type AccountMetadata struct {
	Environment      string  `json:"Environment"`
	OperatorURL      *string `json:"Operator URL"`
	OrganizationID   string  `json:"Organization ID"`
	OrganizationName string  `json:"Organization Name"`
	LocationID       *string `json:"Location ID,omitempty"`
	LocationName     *string `json:"Location Name,omitempty"`
}

// Test service for AccountMetadata shape
type StripePaymentsService interface {
	// Fetch account metadata
	GetAccountMetadata(context.Context, AccountMetadata) (*AccountMetadata, error)
}

// Test service for AccountMetadata shape
type StripePaymentsRPCClient struct {
	lokerpc.Client
}

func (c StripePaymentsRPCClient) GetAccountMetadata(ctx context.Context, req AccountMetadata) (*AccountMetadata, error) {
	var res AccountMetadata
	err := c.DoRequest(ctx, "getAccountMetadata", req, &res)
	if err != nil {
		return nil, err
	}
	return &res, nil
}
